RecoLocalTracker/SiPixelRecHits/interface/pixelCPEforDevice.h

```diff
@@ -122,6 +124,53 @@ namespace pixelCPEforDevice {
     cotbeta = gvy * gvz;
   }
 
+  constexpr inline float check_for_short(uint16_t upper_edge_first_pix,  //!< As the name says.
+                                         uint16_t lower_edge_last_pix,   //!< As the name says.
+                                         float lorentz_shift,            //!< L-shift at half thickness
+                                         float theThickness,             //detector thickness
+                                         float cot_angle,                //!< cot of alpha_ or beta_
+                                         float pitch,                    //!< thePitchX or thePitchY
+                                         bool first_is_big,              //!< true if the first is big
+                                         bool last_is_big)               //!< true if the last is big
+  {
+    //checks if the observed cluster is much shorter than expected
+    //if yes, returns the position predicted using a one-sided reconstruction
+    //if not, returns a dummy value -999.
+    float delta_length_cut = 2.;  //to be fine tuned later
+    float w_inner = pitch * (lower_edge_last_pix - upper_edge_first_pix);
+    float w_pred = theThickness * cot_angle - lorentz_shift;
+
+    float w_eff = std::abs(w_pred) - w_inner;
+
+    float pitchfraction_first = 1.0;
+    float pitchfraction_last = 1.0;
+
+    if (first_is_big)
+      pitchfraction_first += 1.0f;
+    if (last_is_big)
+      pitchfraction_last += 1.0f;
+
+    float sum_of_edge = pitchfraction_first + pitchfraction_last;
+    float delta = w_eff - 0.5 * sum_of_edge * pitch;
+
+    if (delta / pitch > delta_length_cut) {
+      // define the centers of the first and last pixel coordinates
+      float x1 = pitch * (upper_edge_first_pix - 0.5 * pitchfraction_first);
+      float x2 = pitch * (lower_edge_last_pix + 0.5 * pitchfraction_last);
+      //  observed cluster is much shorter than expected, use one-sided reco
+      if (w_pred > 0.f) {
+        float hit_pos = x1 + 0.5 * w_pred;
+        return hit_pos;
+      } else {
+        float hit_pos = x2 + 0.5 * w_pred;
+        return hit_pos;
+      }
```

Comment:
ONLY if you must apply some independent further update to the code, you can save a few cycles by
```suggestion
      //  observed cluster is much shorter than expected, use one-sided reco
      if (w_pred > 0.f) {
        // define the center of the first pixel coordinate
        float x1 = pitch * (upper_edge_first_pix - 0.5 * pitchfraction_first);
        float hit_pos = x1 + 0.5 * w_pred;
        return hit_pos;
      } else {
        // define the center of the last pixel coordinate
        float x2 = pitch * (lower_edge_last_pix + 0.5 * pitchfraction_last);
        float hit_pos = x2 + 0.5 * w_pred;
        return hit_pos;
      }
```